sdk/php/src/Attribute/DefaultPath.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dagger\Attribute;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final readonly class DefaultPath
```

Comment:
You already have a `Dagger\Attribute\Argument`. `DefaultPath` and `Ignore` should be arguments/properties of it. Just like a Function argument's description, they are also metadata for the argument.

Reply:
When we discussed the possibility we felt it was neater to have separate attributes for each.

Reasoning as follows:
1. `DefaultPath` and `Ignore` are not applicable to every `Argument`
2. The convention I've witnessed most commonly in PHP is that each piece of meta data tends to be given a separate attribute.

For example, PHPUnit's attributes look like this:

```php
#[CoversClass(Thing::class)]
#[UsesClass(OtherThing::class)]
class ThingTest extends TestCase
{
    #[Test]
    #[TestDox('description of what is being tested')]
    #[DataProvider('provideTestCases')]
    public function testThingPasses()
    // ...
``` 

Which matches closely to our current implementation: 

```php
#[Argument('Your source directory')]
#[DefaultPath('.')]
#[Ignore('vendor/', 'generated/')]
Directory $source,
```

Reply:
If that's the convention, then you should deprecate `Argument` in favor of `Doc`.

Reply:
> If that's the convention, then you should deprecate `Argument` in favor of `Doc`.

Good point, I agree. Shall I do that in this PR, or make an issue to do in a separate small PR?

Reply:
@helderco I've made a PR to implement this change: https://github.com/dagger/dagger/pull/8486

Reply:
@helderco https://github.com/dagger/dagger/pull/8486 has now been merged.

`Argument` is deprecated (but still functions).

`Doc` is the way to specify a doc-string for both _Dagger Functions_ and _arguments_.

`Argument` is deprecated, but still functional.

If both `Doc` and `Argument` are present: `Doc` is used, `Argument` is ignored.